core/src/main/java/com/opensymphony/xwork2/interceptor/CSRFPreventionInterceptor.java

```diff
@@ -0,0 +1,44 @@
+package com.opensymphony.xwork2.interceptor;
+
+import com.opensymphony.xwork2.ActionContext;
+import com.opensymphony.xwork2.ActionInvocation;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CSRFPreventionInterceptor extends AbstractInterceptor {
+
+    private List<String> exemptedPaths = new ArrayList<String>();
+    private DefaultResourceIsolationPolicy fmPolicy = new DefaultResourceIsolationPolicy();
+
+    public List<String> getExemptedPaths(){
+        return this.exemptedPaths;
+    }
+
+    public void setExemptedPaths(List<String> paths){
+        this.exemptedPaths = paths;
+    }
+
+    @Override
+    public String intercept(ActionInvocation invocation) throws Exception {
+        ActionContext context = invocation.getInvocationContext();
+        HttpServletRequest request = context.getServletRequest();
+
+        fmPolicy.setExemptedPaths(this.exemptedPaths);
+
+        if (!fmPolicy.isRequestAllowed(request)){
+            throw new RuntimeException("Request is not allowed");
```

Comment:
It is unclear what the side effects of this exception are. If the policy rejects the request we need to return a 403 Forbidden response. I suspect this requires getting the servlet response from the `ActionInvocationContext` and setting properties on it, ideally using `javax.servlet.http.HttpServletResponse#SC_FORBIDDEN`. We should of course, return after that, so as to not let the chain of interceptors continue.

Reply:
I did not manage to get a response object from the tests since there is no actual response in tests. I implemented test logic with whether the interceptor.intercept() returns a "403" string or not